addons/AdherentBuffs/AdherentBuffs.lua

```diff
@@ -0,0 +1,112 @@
+--       notice, this list of conditions and the following disclaimer.
+--     * Redistributions in binary form must reproduce the above copyright
+--       notice, this list of conditions and the following disclaimer in the
+--       documentation and/or other materials provided with the distribution.
+--     * Neither the name of AdherentBuffs nor the
+--       names of its contributors may be used to endorse or promote products
+--       derived from this software without specific prior written permission.
+-- 
+-- THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+-- ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+-- WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+-- DISCLAIMED. IN NO EVENT SHALL <your name> BE LIABLE FOR ANY
+-- DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+-- (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+-- LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+-- ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+-- (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+-- SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ 
+config = require ('config')
+ 
+_addon.name     = 'AdherentBuffs'
+_addon.author   = 'Mafai, Sechs'
+_addon.version  = '1.1'
+_addon.commands = {'adherentbuffs','ab'}
+ 
+defaults = T{}
+ 
+defaults.AnnounceMode = 'party' --this can be say/party/linkshell/linkshell2/shout/s/p/l/l2/sh
+defaults.AutoAnnounce = 'off' --this can be true/t/on/false/f/off
+ 
+settings = config.load(defaults)
+ 
+adherent_maps = {['Steadfast Adherent']="PLD, DEF+", ['Furtive Adherent']="WHM, MDB+", ['Occult Adherent']="WAR, EVA+",
+		['Fleet Adherent']="WAR, Haste+", ['Brawny Adherent']="DRK, ATK+", ['Martial Adherent']="DRK,Regain+",
+		['Honed Adherent']="RDM, Fast Cast+", ['Insidious Adherent']="RDM, MEVA+", ['Hexbreaking Adherent']="BLM, MAB+"}
+chatmodes = S{'say','party','linkshell','linkshell2','shout','s','p','l','l2','sh'}
+automodes = S{'true','t','on','false','f','off'}
+moblist = S{}
+		
+windower.register_event('addon command', function (command,...)
+	command = command and command:lower()
+	local args = T{...}
+	if command == 'reload' then
+		windower.send_command('lua unload AdherentBuffs; lua load AdherentBuffs')
+	elseif command == 'unload' then
+		windower.send_command('lua unload AdherentBuffs')
+	elseif command == 'chatmode' or command == 'cm' then
+		if args[1] ~= nil and chatmodes:contains(args[1]) then
+			windower.add_to_chat(038,' ***** Chat Mode changed to "'..args[1]..'" *****')
+			settings.AnnounceMode = args[1]
+			config.save(settings)
+		else
+			windower.add_to_chat(038,' ***** That is not a valid chat mode *****')
+		end
+	elseif command == 'announce' or command == 'a' then
+		announce(1)
+	elseif command == 'autoannounce' or command == 'aa' then
```

Comment:
This section is a perfect example of why `settings.AutoAnnounce` should be a boolean value. Also you should split up the earlier `automodes` table like so:
```lua
false_values = S{'false', 'off', 'f', '0'}
true_values = S{'true', 'on', 't', '1'}
```
Then you can shorten it to this:
```lua
        local value = args[1] and args[1]:lower() or nil
        if not value then
            settings.AutoAnnounce = not settings.AutoAnnounce
        elseif false_values:contains(value) or true_values:contains(value) then
            settings.AutoAnnounce = not false_values:contains(args[1]:lower())
        else
            error('Incorrect syntax: //autoannounce autoannounce [true|false]')
            return
        end

        log('AutoAnnounce setting changed to %s.':format(tostring(settings.AutoAnnounce)))
        config.save(settings)
```